### PR TITLE
Hide secret configuration data

### PR DESCRIPTION
**Pull request by fabxc**
@brian-brazil

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -26,6 +26,16 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+var patAuthLine = regexp.MustCompile(`((?:api_token|api_key|service_key|api_url):\s+)(".+"|'.+'|[^\s]+)`)
```
Comment (brian-brazil) [inline config/config.go]: I don't see an `api_url` in the config

File: config/config.go [hunk 2]
```diff
@@ -26,6 +26,16 @@ import (
...
+
+// Secret is a string that must not be revelead on marshaling.
```
Comment (brian-brazil) [inline config/config.go]: revealed

Comment (fabxc) [inline config/config.go]: Fixed.

**Comment by brian-brazil** [comment]
:+1: 

Having something as generic as `api_url` hidden may come back to bite us as URLs are not generally secure (proxies log them for example).

**Pull request merged**